modules/osdk-control-compat.adoc

```diff
@@ -0,0 +1,87 @@
+====
+
+To prevent workflow issues, cluster administrators must install a version of your Operator in their cluster that is compatible with the next version of {product-title} before they upgrade their cluster from any previous version. While it is recommended that you update your Operators to no longer use deprecated or removed APIs, if you still need to publish your Operator bundles with removed APIs for continued use on earlier versions of {product-title}, ensure that the bundle is configured accordingly.
+
+The following procedure helps prevent administrators from installing versions of your Operator on an incompatible version of {product-title}.  These steps also prevent administrators from upgrading to a newer version of {product-title} that would be incompatible with the version of your Operator that is currently installed on their cluster.
+
+[IMPORTANT]
+====
+You should configure the following settings in your Operator project as soon as possible. Users running workloads with deprecated APIs that are planning to eventually upgrade to a future version of {product-title}, which might remove the deprecated API, must be running an Operator that has the proper configuration to prevent the cluster upgrade and potentially adversely impact their critical workloads.
+====
+
+.Prerequisites
+
+- Operator SDK CLI installed on a development workstation
+- Operator project initialized by using the Operator SDK
+
+.Procedure
+
+. Configure the maximum version of {product-title} that your Operator is compatible with. In your Operator project's cluster service version (CSV), set the `olm.openShiftMaxVersion` annotation to prevent administrators from upgrading their cluster before upgrading the installed Operator to a compatible version:
++
+.Example CSV with `olm.maxOpenShiftVersion` annotation
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    "olm.properties": '[{"type": "olm.maxOpenShiftVersion", "value": "4.8"}]' <1>
+----
+<1> Setting `value` to `4.8` prevents cluster upgrades to {product-title} versions later than 4.8 when this bundle is installed on a cluster.
+
+. Configure the compatible versions of {product-title} for your Operator:
+
+.. Set the `com.redhat.openshift.versions` annotation in your project's `bundle/metadata/annotations.yaml` file:
++
+.Example `bundle/metadata/annotations.yaml` file with compatible versions
+[source,yaml]
+----
+com.redhat.openshift.versions: "v4.6-v4.8" <1>
+----
+<1> Set to a range or single version.
+
+.. To prevent your bundle from being carried on to an incompatible version of {product-title}, ensure that the index image is generated with the proper `com.redhat.openshift.versions` label in your project's `bundle.Dockerfile`:
++
+.Example `bundle.Dockerfile` with compatible versions
++
+[source,yaml]
+----
+LABEL com.redhat.openshift.versions="v4.6-v4.8" <1>
```

Comment:
Only I nit. Unfortunately, we need to say here add to the annotations file and also ensure the sync for the bundle. Dockerfile. Something such as; 

```
Use the annotation `com.redhat.openshift.versions` in `bundle/metadata/annotations.yaml` and ensure that the index image (`bundle.Dockerfile`) is generated with its OCP Label, to prevent the bundle from being carried on to 4.9, e.g.:

In `bundle/metadata/annotations.yaml`:

`com.redhat.openshift.versions: "v4.6-v4.8"`

In `bundle.Dockerfile`

`LABEL com.redhat.openshift.versions="v4.6-v4.8"`

```